Test/Invoke-WhiskeyTask.Tests.ps1

```diff
@@ -584,36 +617,30 @@ function WhenRunningTask
         [string]
         $Name,
 
-        [hashtable]
-        $Parameter,
-
         [string]
         $InRunMode
     )
 
     Mock -CommandName 'Invoke-PreTaskPlugin' -ModuleName 'Whiskey'
     Mock -CommandName 'invoke-PostTaskPlugin' -ModuleName 'Whiskey'
 
-    $byItDepends = @{ 'ForDeveloper' = $true }
-    if( $runByBuildServer )
+    $script:context.PipelineName = 'Build'
+    $script:context.TaskName = $null
+    $script:context.TaskIndex = 1
+    
+    if ($taskDefaults.Keys -gt 0)
     {
-        $byItDepends = @{ 'ForBuildServer' = $true }
+        $script:context.TaskDefaults = $taskDefaults
     }
 
-    $script:context = New-WhiskeyTestContext @byItDepends
-    $context.PipelineName = 'Build';
-    $context.TaskName = $null;
-    $context.TaskIndex = 1;
-    $context.TaskDefaults = $taskDefaults;
-
     if( $InRunMode )
     {
-        $context.RunMode = $InRunMode;
+        $script:context.RunMode = $InRunMode

```

Comment:
You only need the `$script:` scope when setting the value of the variable, not properties on the variable.